Makefile

```diff
@@ -363,25 +363,40 @@ release-build: bundle image-build bundle-build catalog-build ## Build operator,
 .PHONY:
 release-push: image-push bundle-push catalog-push ## Push operator, bundle + catalog images
 
+# It has to be the same namespace as ./config/default/kustomization.yaml -> namespace
+OPERATOR_NAMESPACE ?= backstage-system
+OLM_NAMESPACE ?= olm
+
 .PHONY: deploy-olm
 deploy-olm: ## Deploy the operator with OLM
-	kubectl apply -f config/samples/catalog-operator-group.yaml
-	sed "s/{{VERSION}}/$(subst /,\/,$(VERSION))/g" config/samples/catalog-subscription-template.yaml | sed "s/{{DEFAULT_OLM_NAMESPACE}}/$(subst /,\/,$(DEFAULT_OLM_NAMESPACE))/g" | kubectl apply -f -
+	kubectl -n ${OPERATOR_NAMESPACE} apply -f config/samples/catalog-operator-group.yaml
+	sed "s/{{VERSION}}/$(subst /,\/,$(VERSION))/g" config/samples/catalog-subscription-template.yaml | sed "s/{{OLM_NAMESPACE}}/$(subst /,\/,$(OLM_NAMESPACE))/g" | kubectl -n ${OPERATOR_NAMESPACE} apply -f -
 
 .PHONY: undeploy-olm
 undeploy-olm: ## Un-deploy the operator with OLM
-	-kubectl delete subscriptions.operators.coreos.com backstage-operator
-	-kubectl delete operatorgroup backstage-operator-group
-	-kubectl delete clusterserviceversion backstage-operator.$(VERSION)
+	-kubectl -n ${OPERATOR_NAMESPACE} delete subscriptions.operators.coreos.com backstage-operator
+	-kubectl -n ${OPERATOR_NAMESPACE} delete operatorgroup backstage-operator-group
+	-kubectl -n ${OPERATOR_NAMESPACE} delete clusterserviceversion backstage-operator.v$(VERSION)
 
-DEFAULT_OLM_NAMESPACE ?= openshift-marketplace
 .PHONY: catalog-update
 catalog-update: ## Update catalog source in the default namespace for catalogsource
-	-kubectl delete catalogsource backstage-operator -n $(DEFAULT_OLM_NAMESPACE)
-	sed "s/{{CATALOG_IMG}}/$(subst /,\/,$(CATALOG_IMG))/g" config/samples/catalog-source-template.yaml | kubectl apply -n $(DEFAULT_OLM_NAMESPACE) -f -
+	-kubectl delete catalogsource backstage-operator -n $(OLM_NAMESPACE)
+	sed "s/{{CATALOG_IMG}}/$(subst /,\/,$(CATALOG_IMG))/g" config/samples/catalog-source-template.yaml | kubectl apply -n $(OLM_NAMESPACE) -f -
 
+# Deploy on Openshift cluster using OLM (by default installed on Openshift)
 .PHONY: deploy-openshift
-deploy-openshift: release-build release-push catalog-update ## Deploy the operator on openshift cluster
+deploy-openshift: OLM_NAMESPACE=openshift-marketplace release-build release-push catalog-update deploy-olm ## Deploy the operator on openshift cluster
+
+.PHONY: install-olm
+install-olm: operator-sdk
+	$(OPSDK) olm install
+
+.PHONY: create-operator-namespace
+create-operator-namespace:
+	-kubectl create namespace ${OPERATOR_NAMESPACE}
+
+.PHONY: deploy-k8s-olm
+deploy-k8s-olm: release-build release-push catalog-update create-operator-namespace deploy-olm ## Deploy the operator on openshift cluster
```

Comment:
Why do we have this additional `deploy-k8s-olm`, while the existing `deploy-olm` seems to be sufficient?
Also, I am not sure we need to depend on `release-build` and `release-push`, as this would require the ability to push to a registry. But what if I just want to deploy using the default upstream images (from `quay.io/janus-idp`)?
I think `release-build` and `release-push` should run separately upon request.

Reply:
I consider it as a helper for developer, to not to keep in the brain all the steps needed, same as deploy-openshift. 
If one want to perform only certain steps - nothing prevents to. No?

Reply:
Yeah, but I find it redundant (and a bit confusing).. Plus, the `release-push` target requires push access to a registry.
If I just cloned the repo, and just want to deploy the operator in my cluster (where OLM is already installed) and I don't have access to a registry, which command should I run? `deploy-olm` or `deploy-k8s-olm`?
To me, only `make catalog-update deploy-olm`. (And that's why I was even suggesting that `deploy-olm` depends on `catalog-update`, so that people can just run `make deploy-olm` on K8s and have it working with the operator upstream images).
IMO, `release-build` and `release-push` should not be needed to deploy with OLM.

Reply:
I believe, you, as developer, should have access to "some" registry to be able to test the full cycle, no?

As it stated in the doc "To build and deploy the operator... with OLM", so if you need build and deploy - use it, if you need just deploy - use  `catalog-update` and|or `deploy-olm` or whatever.
I find `deploy-k8s-olm` useful because I do not do it that often and I do not want to remember all the sequence every time.

Regarding whether we need separated (or dependant) `deploy-olm` and `catalog-update`, I have no strong opinion. In this case I'd just use pre-created options and see what to improve in the future.

Reply:
> I believe, you, as developer, should have access to "some" registry to be able to test the full cycle, no?

That makes sense if I am iterating locally and want to test everything, I agree. But I was more referring to someone who just cloned the repo and just wants to deploy using the upstream images..
Similarly to how `make deploy` just works, `make deploy-olm` should do the same thing. And if I want to build and deploy from my own images, I can run `make release-build release-push deploy-olm IMG=[my-registry/my-repo]`, but separately.
This is why I found this new `deploy-k8s-olm` to be kinda confusing.


> As it stated in the doc "To build and deploy the operator... with OLM", so if you need build and deploy - use it, if you need just deploy - use catalog-update and|or deploy-olm or whatever.
I find deploy-k8s-olm useful because I do not do it that often and I do not want to remember all the sequence every time.

In this case, maybe you could call this target differently (with `build` somewhere in the name). `deploy-k8s-olm` looks really confusing compared to `deploy-olm` ;)

> Regarding whether we need separated (or dependant) deploy-olm and catalog-update, I have no strong opinion. In this case I'd just use pre-created options and see what to improve in the future.

The current `deploy-olm` target only deploys the OperatorGroup and Subscription, which won't be useful until the catalog source is installed with `make catalog-update`. That's why I thought it would really help to add this dependency. But okay, I'm fine with what we have now.

To summarize, I think:

- `deploy-olm` is good as it is if it also depends on `catalog-update` to install the catalog source. So I am not sure we need an additional `deploy-k8s-olm` target.
- we can have a `deploy-openshift` target as helper, and the only difference would be that it would set the OLM namespace and do exactly the same thing as `deploy-olm`. No need at all to depend on the `release-*` targets, IMO.
- I am not sure we need an additional `deploy-olm-openshift`, which I also find confusing. If we have the `deploy-openshift` target above, it implies that we'll be using the OpenShift way for deploying the operator. Otherwise, there is already the `deploy` target handy.
- Not sure either we need an additional `catalog-update-openshift` target. If `deploy-openshift` just sets the expected OLM namespace and does the same thing as `deploy-olm`

This way, users can run:
- `make deploy-olm` or `make deploy-openshift` to deploy using OLM
- they can call `make release-build release-push` on demand if they need to build a different image. Then call `make deploy-olm`, `make deploy-openshift` or `make deploy` depending on what they want to test

Does that make sense?

Reply:
I added and documented what I found useful for developer, mostly to not to try to remember every time all the specifics.
I did not consider redundancy as a problem, to me makefile is a set of helpers.
For the deploy-olm dependency on catalog-update , like I said, I changed nothing, it is not the point of this PR. 

I believe all the "older" things should work as before.
If you see any breaking changes or documentation inconsistency, please let me know and I'll fix it ASAP, otherwise I would propose to merge it and, of course, you can modify and propose yours, more elegant, version of makefile and documentation as a separate PR.
Again, my point is to make it possible to take a glance on doc and make what I, as developer, need, e2e. 

Does it make sense?